internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_availability_set_resource.go

```diff
@@ -0,0 +1,210 @@
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for the presence of an existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			parameters := availabilitysets.AvailabilitySet{
+				Location: location.Normalize(model.Location),
+				ExtendedLocation: availabilitysets.ExtendedLocation{
+					Type: utils.String("customLocation"),
+					Name: utils.String(model.CustomLocationId),
+				},
+				Properties: availabilitysets.AvailabilitySetProperties{
+					AvailabilitySetName: utils.String(id.AvailabilitySetName),
+					VMmServerId:         utils.String(scvmmServerId.ID()),
+				},
+				Tags: pointer.To(model.Tags),
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, parameters); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r SystemCenterVirtualMachineManagerAvailabilitySetResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.SystemCenterVirtualMachineManager.AvailabilitySets
+
+			id, err := availabilitysets.ParseAvailabilitySetID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(*id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := SystemCenterVirtualMachineManagerAvailabilitySetModel{}
+			if model := resp.Model; model != nil {
+				state.Name = id.AvailabilitySetName
+				state.ResourceGroupName = id.ResourceGroupName
+				state.Location = location.Normalize(model.Location)
+				state.CustomLocationId = pointer.From(model.ExtendedLocation.Name)
+				state.Tags = pointer.From(model.Tags)
+
+				scvmmServerId, err := vmmservers.ParseVMmServerID(pointer.From(model.Properties.VMmServerId))
+				if err != nil {
+					return err
+				}
+				state.SystemCenterVirtualMachineManagerServerId = scvmmServerId.ID()
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r SystemCenterVirtualMachineManagerAvailabilitySetResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.SystemCenterVirtualMachineManager.AvailabilitySets
+
+			id, err := availabilitysets.ParseAvailabilitySetID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model SystemCenterVirtualMachineManagerAvailabilitySetModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			parameters := availabilitysets.ResourcePatch{
+				Tags: pointer.To(model.Tags),
```

Comment:
Though currently only `tags` can be updated. other updatable properties may be added in future. Shall we still use `if d.HasChange("tags")` here?

Reply:
Updated